docs/hello_nextflow/03_hello_plumbing.md

```diff
@@ -0,0 +1,755 @@
+
+Good news: we can declare as many input variables as we want. Let's call this one `batch_name`.
+
+In the process block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="48"
+    input:
+        path input_files
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="48"
+    input:
+        path input_files
+        val batch_name
+```
+
+### 3.2. Use the `batch_name` variable in the output file name
+
+In the process block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="52"
+    output:
+        path "COLLECTED-output.txt"
+
+    script:
+    """
+    cat ${input_files} > 'COLLECTED-output.txt'
+    """
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="52"
+    output:
+        path "COLLECTED-${batch_name}-output.txt"
+
+    script:
+    """
+    cat ${input_files} > 'COLLECTED-${batch_name}-output.txt'
+    """
+```
+
+This sets up the process to use the `batch_name` value to generate a specific filename for the final output of the workflow.
+
+### 3.3. Add a `batch` command-line parameter
+
+Now we need a way to supply the value for `batch_name`.
+
+You already know how to use the `params` system to declare CLI parameters.
+Let's use that to declare a `batch` parameter (with a default value because we are lazy).
+
+In the pipeline parameters section, make the following code changes:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="61"
+/*
+ * Pipeline parameters
+ */
+params.greeting = 'data/greetings.csv'
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="61"
+/*
+ * Pipeline parameters
+ */
+params.greeting = 'data/greetings.csv'
+params.batch = 'test-batch'
+```
+
+Remember you can override that default value by specifying a value with `--batch` on the command line.
+
+### 3.4. Pass the `batch` parameter to the process
+
+To provide the value of the parameter to the process, we need to add it in the process call.
+
+In the workflow block, make the following code change:
+
+_Before:_
+
+```groovy title="hello-plumbing.nf" linenums="80"
+    // collect all the greetings into one file
+    collectGreetings(convertToUpper.out.collect())
+```
+
+_After:_
+
+```groovy title="hello-plumbing.nf" linenums="80"
+    // collect all the greetings into one file
+    collectGreetings(convertToUpper.out.collect(), params.batch)
+```
+
+!!! warning
+
+    You MUST provide the inputs to a process in the EXACT SAME ORDER as they are listed in the input definition block of the process.
+
+### 3.5. Run the workflow
+
+Let's try running this with a batch name on the command line.
+
+```bash
+nextflow run hello-plumbing.nf -resume --batch trio
+```
+
+It runs successfully and produces the desired output:
+
+```console title="bash"
+cat results/COLLECTED-trio-output.txt
+```
+
+```console title="Output"
+HELLO
+BONJOUR
+HOLà
+```
+
+### Takeaway
+
+You know how to pass more than one input through a channel.
```

Comment:
```suggestion
You know how to pass collected inputs through a channel.
```

To dissambuguate from the earlier examples with multiple items.

Reply:
I find this _more_ ambiguous though; this reads like it's about the collected greetings, but it's not; it's about that parameter that constitutes an extra 'track' in the input channel. This is exactly what I was talking about in my other comment just now; the ways we talk about multiple inputs is inconsistent and confusing.

Reply:
To me 'passing more than one input through a channel' covers both things- you're doing it in both cases. The difference here is that we're passing multiple inputs to a single task (a task being a single iteration of a process).

Reply:
But that's not what this specific line is about; that's not the takeaway. The takeaway from this specific section is that we now know that if the process expects `path foo` and `val bar`, i have to write the process call as `myProcess(foo, bar)`. Before this, all I knew to do was `myProcess(foo)`.

Reply:
OK, apologies for the quick comment, I conflated issues while multitasking and I've re-read. 

In fact, the point of this section (section 3) is that we've added a second input to the process, and filled that from a new parameter, right? So both of our versions are wrong. It's nothing to do with multiple inputs through 'a' channel (in fact there are now 2 channels, though one is implicit).

Maybe something like:

```
You know how to add more than one input channel to a process, and use both inputs from the workflow
```

Reply:
Updated this in a previous pass, thanks for clarifying